### PR TITLE
adds support for pointers to attributes on remote tensors

### PR DESCRIPTION
**Pull request by iamtrask**
Also adds automatic support for .grad pointer

### CODE REVIEW

File: syft/frameworks/torch/tensors/pointer.py [hunk 1]
```diff
@@ -127,16 +128,20 @@ def __str__(self):
...
-            f"id@loc:{self.id_at_location}"
+            # f"id:{self.id} "
+            # f"owner:{self.owner.id} "
+            f"...{str(self.id_at_location)[-3:]}@{self.location.id}"
+            # f"id@loc:"
             f"]"
         )
 
+        if(self.point_to_attr is not None):
+            out += "::" + str(self.point_to_attr)
```
Comment (LaRiffle) [inline syft/frameworks/torch/tensors/pointer.py]: maybe you want to do `'::'.join(self.point_to_attr.split('.')`

File: test/torch/tensors/test_logging.py [hunk 2]
```diff
@@ -109,7 +109,7 @@ def test_remote_method_on_log_chain(self):
         x_ptr = x.send(self.bob)
         y_ptr = F.relu(x_ptr)
         y = y_ptr.get()
-        assert (y.child.child == F.relu(x_tensor)).all()
+        assert (y.child == F.relu(x_tensor)).all()
```
Comment (LaRiffle) [inline test/torch/tensors/test_logging.py]: type(y) : wrapper>logging>tensor

File: syft/frameworks/torch/tensors/pointer.py (hunk 1 above)
Comment (iamtrask) [reply syft/frameworks/torch/tensors/pointer.py]: fixed

File: test/torch/tensors/test_pointer.py [hunk 3]
```diff
@@ -171,3 +171,38 @@ def test_gradient_send_recv(self):
...
+        assert isinstance(y, torch.Tensor)
+        assert isinstance(y.child, syft.LoggingTensor)
+        assert isinstance(y.child.child, torch.Tensor)
+
+        # call .get() on pinter to attribute (should error)
+        x.child.point_to_attr = "child"
+        try:
+            x.get()
+        except syft.exceptions.CannotRequestTensorAttribute as e:
+            assert True
```
Comment (LaRiffle) [inline test/torch/tensors/test_pointer.py]: 👍

File: test/torch/tensors/test_logging.py (hunk 2 above)
Comment (iamtrask) [reply test/torch/tensors/test_logging.py]: fixed

File: syft/frameworks/torch/tensors/pointer.py [hunk 4]
```diff
@@ -119,16 +128,20 @@ def __str__(self):
...
-        return (
+        out = (
             f"["
             f"{type_name} - "
-            f"id:{self.id} "
-            f"owner:{self.owner.id} "
-            f"loc:{self.location.id} "
-            f"id@loc:{self.id_at_location}"
+            # f"id:{self.id} "
+            # f"owner:{self.owner.id} "
```
Comment (robert-wagner) [inline syft/frameworks/torch/tensors/pointer.py]: Remove comments or uncomment

Comment (iamtrask) [reply syft/frameworks/torch/tensors/pointer.py]: fixed

**Comment by robert-wagner** [comment]
Please wait in the future for Travis to pass before merging. It prevents dumb errors

**Pull request merged**